.github/workflows/ci.yml

```diff
@@ -6,77 +6,76 @@ on:
-      - id: get-user
-        run: echo "uid_gid=$(id -u):$(id -g)" >> "$GITHUB_OUTPUT"
-  build-and-deploy:
-    needs: get-runner-user
+  build-and-release:
     runs-on: ubuntu-22.04
-    container:
-      image: "public.ecr.aws/z8l5l4v4/buildtools/tool/maven:3-jdk-11-debian"
-      options: --user ${{ needs.get-runner-user.outputs.uid_gid }}
     permissions:
       contents: write
       packages: write
     steps:
       - uses: actions/checkout@v4
-      - uses: actions/cache@v3
+
+      - uses: actions/setup-java@v4
+        with:
+          distribution: "zulu"
+          java-version: "11.0.22"
+          java-package: "jdk"
+
+      - uses: capralifecycle/actions-lib/check-runtime-dependencies@v1
+
+      - name: cache mvn
+        uses: actions/cache@v4
         with:
           path: ~/.m2/repository
-          key: maven-${{ hashFiles('**/pom.xml') }}
-          restore-keys: maven-
-      - name: store branch name in environment
-        run: echo "BRANCH_NAME=$(echo ${GITHUB_REF##*/})" >> "$GITHUB_ENV"
-      - name: git config
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-
+
+      - name: release preparation
+        id: prep
         run: |
           git config --global user.name "github-actions[bot]"
           git config --global user.email "github-actions[bot]@users.noreply.github.com"
-      - name: configure maven
-        run: |
-          mkdir -p "$HOME/.m2"
-          cat <<EOF > "$HOME/.m2/settings.xml"
-          <settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
-            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-            xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
-            <servers>
-              <server>
-                <id>github</id>
-                <username>\${env.GITHUB_ACTOR}</username>
-                <password>\${env.GITHUB_TOKEN}</password>
-              </server>
-            </servers>
-          </settings>
-          EOF
+          major_version="$(sed -n "s/^.*<major-version>\([0-9]\{1,\}\)<\/major-version>.*$/\1/p" pom.xml)"
+          echo "$major_version" | grep -q "^[0-9]\{1,\}$" || {
+            echo "ERROR: Failed to extract <major-version> from pom.xml"
+            exit 1
+          }
+          echo "major-version=$major_version" >> "$GITHUB_OUTPUT"
+
+      - uses: capralifecycle/actions-lib/generate-tag@v1
+        id: tag
+        with:
+          tag-prefix: "${{ steps.prep.outputs.major-version }}"
+          tag-type: "punctuated-timestamp-tag"
+
+      - name: resolve dependencies
+        env:
+          # NOTE: Use the following token if your project does not depend on private packages:
+          # GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # NOTE: Use the following token if your project depends on private packages:
+          GITHUB_TOKEN: ${{ secrets.SHARED_GITHUB_PACKAGES_TOKEN }}
```

Comment:
Not needed - the built-in GITHUB_TOKEN should suffice.

Reply:
https://github.com/capralifecycle/liflig-properties/pull/57/commits/0c2cc46e464f194d04c9719e192b811c76f86e40